Plugin/src/main/java/xyz/kyngs/librelogin/common/listener/LoginTryListener.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package xyz.kyngs.librelogin.common.listener;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import xyz.kyngs.librelogin.api.event.events.AuthenticatedEvent;
+import xyz.kyngs.librelogin.api.event.events.WrongPasswordEvent;
+import xyz.kyngs.librelogin.api.event.events.WrongPasswordEvent.AuthenticationSource;
+import xyz.kyngs.librelogin.common.AuthenticLibreLogin;
+import xyz.kyngs.librelogin.common.config.ConfigurationKeys;
+
+public class LoginTryListener<P, S> {
+
+    private final AuthenticLibreLogin<P, S> plugin;
+    private final Map<P, Integer> loginTries = new ConcurrentHashMap<>();
+
+    public LoginTryListener(AuthenticLibreLogin<P, S> libreLogin) {
+        this.plugin = libreLogin;
+        libreLogin.getEventProvider().subscribe(libreLogin.getEventTypes().wrongPassword, this::onWrongPassword);
+        libreLogin.getEventProvider().subscribe(libreLogin.getEventTypes().authenticated, this::onAuthenticated);
+    }
+
+    private void onWrongPassword(WrongPasswordEvent<P, S> wrongPasswordEvent) {
+        if (wrongPasswordEvent.getSource() != AuthenticationSource.LOGIN)
+            return;
+        // if key do not exists, put 1 as value
+        // otherwise sum 1 to the value linked to key
+        loginTries.merge(wrongPasswordEvent.getPlayer(), 1, Integer::sum);
+        if (loginTries.getOrDefault(wrongPasswordEvent.getPlayer(), 0) >= plugin.getConfiguration().get(ConfigurationKeys.MAX_LOGIN_TRIES)) {
+            plugin.getPlatformHandle().kick(wrongPasswordEvent.getPlayer(), plugin.getMessages().getMessage("kick-too-many-login-tries"));
+        }
+    }
+
+    private void onAuthenticated(AuthenticatedEvent<P, S> authenticatedEvent) {
+        loginTries.remove(authenticatedEvent.getPlayer());
```

Comment:
What if the player disconnects while they're in the limbo? Wouldn't that keep the player on the map indefinitely?

Reply:
Yes, you are right.
I thought about that, and I did this because people can "brute" by simply rejoining the server.
Or is this too complicated and we should just remove them from the map when player leave from the server?